packages/ipfs-http-client/src/lib/multipart-request.js

```diff
@@ -49,6 +49,9 @@ async function multipartRequest (source, boundary = `---------------------------
 
         index++
       }
+    } catch (err) {
+      // workaround for https://github.com/node-fetch/node-fetch/issues/753
+      abortController.abort(err)
```

Comment:
the change you made to `http` makes this error surface when we do?
```js
try {
api.post()
}catch (err){
	//here?
}

```
can't we pass the http instance instead of a new abortController and abort with the internal controller?

Reply:
this would avoid having to do 
```
// allow aborting requests on body errors
    const controller = new AbortController()
    const signal = anySignal([controller.signal, options.signal]
````
everywhere

Reply:
I'm not sure I follow, could you explain a bit more?

We need to create a new abort controller in `ipfs-utils/src/http` to support timeouts, and we can't reuse abort controller instances (once their signal is aborted, it's aborted forever) so using it in the next request makes the request abort immediately.

Reply:
in this PR https://github.com/ipfs/js-ipfs-utils/pull/41/files each request has it's own abort controller, so with that and changing multipart-request to receive the http instance we can call the internal abort controller, `http.abortController.abort()`
makes sense?